src/server/db_slice.cc

```diff
@@ -484,7 +483,8 @@ OpResult<DbSlice::ItAndUpdater> DbSlice::FindMutable(const Context& cntx, string
 
 OpResult<DbSlice::ItAndUpdater> DbSlice::FindMutableInternal(const Context& cntx, string_view key,
                                                              std::optional<unsigned> req_obj_type) {
-  auto res = FindInternal(cntx, key, req_obj_type, UpdateStatsMode::kMutableStats);
+  SlotId key_slot = KeySlot(key);
```

Comment:
we calculate the key slot on transaction init.
How about passing it in the cntx instead of calculating it here again?